### PR TITLE
Componentise the event box

### PR DESCRIPTION
**Pull request by peteryates**
### JIRA ticket number

N/A

### Context

The event box and short event boxes contain some logic in the template that could be more easily tested if it was part of a component.

### Changes proposed in this pull request

Convert the `_eventbox` and `_shorteventbox` partial into a [ViewComponent](https://github.com/github/view_component) and add some more tests of its functionality.

### Guidance to review

I've tried not to change any logic and leave the templates alone.

Some things I noticed along the way:

* the HTML class-building in the template (`class="event-resultbox__content__icon icon-<%= type_name.parameterize %>"`) doesn't account for a unknown or irretrievable type name. Is it guaranteed to be there?

* there appear to be two ways of classifying an event as 'Online', its `isOnline`/`is_online` flag and if its type is `'Online Event'`. Is there a possibility events will have an `is_online: true` _and_ be a type of event other than `Online Event`? Is the flag superflous?

* when an unrecognised or missing event type is provided the call to `t` can't find a translation and it emits a class name like `event-resultbox__divider--span-class-translation_missing-title-translation-missing-en-event_types-name-singular-singular-span`. This doesn't appear to have any side effects but should probably be tidied up

### CODE REVIEW

File: app/components/events/event_box_short_component.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module Events
+  # EventBoxShortComponent is just a smaller version of EventBoxComponent
```
[the remark this answers is not in the document]
Comment (peteryates) [reply app/components/events/event_box_short_component.rb:2]: Yeah - I was planning on just changing as-is in the first pass, then going back and condensing the logic and moving it from the template to the component and combining them in the second. They definitely feel like they should be variants of the same thing.

File: app/components/events/event_box_component.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    def formatted_location
+      safe_format(location)
+    end
+
+    # FIXME: move all the logic that determines whether this
+    #        event is online here. There are currently two
+    #        distinct ways, the is_online flag and checking
+    #        whether the event type's name is 'Online Event'
+    def online?
+      online
```
[the remark this answers is not in the document]
Comment (peteryates) [reply app/components/events/event_box_component.rb:37]: Yeah I was a bit thrown by the HTML here, `<h5>` doesn't feel like the right tag semantically now I've had a look at how the boxes look on screen. The idea that two headings could be rendered one after the other made me think it should be either/or.

[the remark this answers is not in the document]
Comment (ethax-ross) [reply app/components/events/event_box_component.rb:37]: Yeah I think that would be cleaner and get rid of the `Online Event` type check, so you could just have:

```
if online?
  <render online badge>
else
  <render type badge>
end
```

I think the `FIXME` can be removed as well - an event with category `Online Event` should always have `is_online` set to `true`, but if you wanted to belt and braces it you could check for that type in the method as well.

[the remark this answers is not in the document]
Comment (peteryates) [reply app/components/events/event_box_component.rb:33]: I've set this back to how it was before and written some more tests to (hopefully) make the behaviour clearer

**Review by ethax-ross — APPROVED** [review]
LGTM

**Pull request merged**